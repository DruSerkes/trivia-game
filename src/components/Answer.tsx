import * as React from 'react';
import { ListItem } from '@material-ui/core'
import { showIfCorrect } from '../helpers/addClass'

type AnswerProps = {
    answer: string;
    answerQuestion: (answer: string) => void;
    correct?: boolean | null;
}

export function Answer({ answer, answerQuestion, correct }: AnswerProps) {
    const renderHTML = () => (<span className="AnswerText" dangerouslySetInnerHTML={{ __html: answer }} />)
    const handleClick = ({ nativeEvent: e }: React.MouseEvent) => {
        showIfCorrect(e, correct);
        setTimeout(() => answerQuestion(answer), 600);
    }
    return (
        <ListItem id={answer} className="Answer" divider onClick={handleClick}>{renderHTML()}</ListItem>
    );
}