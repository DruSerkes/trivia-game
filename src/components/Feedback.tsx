import * as React from 'react';



export const Feedback = () => (< p className="Feedback" data-testid="Feedback" />)