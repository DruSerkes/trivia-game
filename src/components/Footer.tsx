import React from "react";

export const Footer = () => {
  const year = new Date().getFullYear();
  return (
    <footer className="Footer">
      <p>
        <a
          data-testid="footer"
          href="https://github.com/druserkes"
          target="_blank"
          rel="noopener noreferrer"
        >
          Dru Serkes &copy;{year}
        </a>
      </p>
    </footer>
  );
};
