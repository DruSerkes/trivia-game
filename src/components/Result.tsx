import * as React from 'react';
import { Box, Typography } from '@material-ui/core'
import { renderColor } from '../helpers/renderColor'

type ResultTypes = {
    numCorrect: number;
    numAnswered: number;
}

export const Result = ({ numCorrect, numAnswered }: ResultTypes) => {
    return (
        <Box className="Result">
            <Typography variant="h3">Game Over!</Typography>
            <Typography variant="body1" style={{ color: renderColor(numCorrect) }}>You answered {numCorrect} out of {numAnswered} correct</Typography>
        </Box>
    )
}