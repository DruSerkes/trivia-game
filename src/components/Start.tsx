import * as React from 'react';
import { useState } from 'react';
import { Box, Button, TextField, FormControl, RadioGroup, FormControlLabel, Radio, FormLabel } from '@material-ui/core'

type StartProps = {
    startGame: (amount: string, difficulty: string, type: string) => void
}

export const Start = ({ startGame }: StartProps) => {
    const [amount, setAmount] = useState<string>('10');
    const [difficulty, setDifficulty] = useState<string>('');
    const [type, setType] = useState<string>('multiple');

    const handleChangeAmount = (e: React.ChangeEvent<HTMLInputElement>): void => setAmount(e.target.value);
    const handleChangeDifficulty = (e: React.ChangeEvent<HTMLInputElement>): void => setDifficulty(e.target.value);
    const handleChangeType = (e: React.ChangeEvent<HTMLInputElement>): void => setType(e.target.value);
    const handleSubmit = () => startGame(amount, difficulty, type);

    return (
        <Box className="Start">
            <TextField label="Number of Questions" type="number" value={amount} onChange={handleChangeAmount} />
            <FormControl component="fieldset">
                <FormLabel component="legend">Difficulty</FormLabel>
                <RadioGroup row aria-label="difficulty" name="difficulty" value={difficulty} onChange={handleChangeDifficulty}>
                    <FormControlLabel value="easy" control={<Radio />} label="Easy" labelPlacement="end" />
                    <FormControlLabel value="medium" control={<Radio />} label="Medium" labelPlacement="end" />
                    <FormControlLabel value="hard" control={<Radio />} label="Hard" labelPlacement="end" />
                    <FormControlLabel value="" control={<Radio />} label="Surprise Me!" labelPlacement="end" />
                </RadioGroup>
            </FormControl>
            <FormControl component="fieldset">
                <FormLabel component="legend">Type</FormLabel>
                <RadioGroup row aria-label="type" name="type" value={type} onChange={handleChangeType}>
                    <FormControlLabel value="multiple" control={<Radio />} label="Multiple Choice" labelPlacement="end" />
                    <FormControlLabel value="boolean" control={<Radio />} label="True/False" labelPlacement="end" />
                </RadioGroup>
            </FormControl>
            <Button variant="contained" fullWidth color="primary" onClick={handleSubmit}>Start New Game</Button>
        </Box >
    )
}