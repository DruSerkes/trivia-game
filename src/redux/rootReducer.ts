import { questionsReducer } from './questionsReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ questions: questionsReducer });

export default rootReducer;
