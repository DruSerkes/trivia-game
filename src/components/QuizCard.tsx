import * as React from 'react';
import { QuestionType } from '../redux/types';

type QuizCardProps = {
    question: QuestionType | null;
    answerQuestion: (answer: string) => void
}


export const QuizCard = ({ question, answerQuestion }: QuizCardProps) => {
    // Render the 3 incorrect answers and the correct answer
    // Do so with each being a separate Answer component
    // Answer component should accept props for question and isCorrect (optional bool)
    // When Answer is clicked, it should fire off answerQuestion with the answer passed in


    return (
        <div className="QuizCard">
            <h4>{question?.question}</h4>
            {/* RENDER 4 ANSWERS HERE */}
        </div>
    )
}