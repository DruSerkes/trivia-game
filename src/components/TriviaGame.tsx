/*
    * Redux lives here 
    * Local state to keep track of current question (which gets passed down) 
    * Local state to keep track of answers (correct, answered)  
    * Renders Start if no questions in state
    * Renders QuizCard otherwise
        * pass question data down 
    * Renders Result if â€¦
    * 
    * 
    * res.data.results[0-9]
        * set amount=10
        * set type=multiple
        * .question 
        * .correct_answer 
        * .incorrect_answers[0-2]
*/
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { StoreType } from '../redux/store';
import { QuestionType } from '../redux/types';

// TODO
// 1. Create QuizCard component
// 2. Create Result component
// 3. import each here

const TriviaGame: React.FC = () => {
	const questions = useSelector((state: StoreType) => state.questions);
	const [ currentQuestion, setCurrentQuestion ] = useState<QuestionType | null>(null);
	const [ numAnswered, setNumAnswered ] = useState(0);
	const [ numCorrect, setNumCorrect ] = useState(0);

	const answerQuestion = (answer: string) => {
		if (currentQuestion && answer === currentQuestion.correct_answer) setNumCorrect((correct) => correct++);
		setNumAnswered((answered) => answered++);
		setCurrentQuestion(() => questions[numAnswered]);
	};

	return (
		<div className="TriviaGame">
			{numAnswered < 10 ? (
				<QuizCard question={currentQuestion} answerQuestion={answerQuestion} />
			) : (
				<Result numCorrect={numCorrect} />
			)}
		</div>
	);
};

export default TriviaGame;
