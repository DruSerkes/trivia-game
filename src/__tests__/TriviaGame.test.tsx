import * as React from 'react';
import { render, fireEvent, waitForDomChange } from '@testing-library/react'
import { TriviaGame } from '../components/TriviaGame'
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../redux/rootReducer'
import { composeWithDevTools } from 'redux-devtools-extension';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));


describe('TriviaGame tests', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <TriviaGame />
            </Provider>
        )
        expect(asFragment()).toMatchSnapshot()
    })
})


test('Clicking the button starts the game ', async () => {
    const { getByText, getByTestId } = render(
        <Provider store={store}>
            <TriviaGame />
        </Provider>);

    const startBtn = getByText('Start New Game');
    fireEvent.click(startBtn);
    await waitForDomChange()
    expect(startBtn).not.toBeInTheDocument();
    const quizCard = getByTestId('QuizTest');
    expect(quizCard).toBeInTheDocument();
});