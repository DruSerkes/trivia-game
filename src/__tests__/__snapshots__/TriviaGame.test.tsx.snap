// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaGame tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1 TriviaGame"
  >
    <div
      class="MuiBox-root MuiBox-root-2 Start"
    >
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink="true"
        >
          Number of Questions
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            type="number"
            value="10"
          />
        </div>
      </div>
      <fieldset
        class="MuiFormControl-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          Difficulty
        </legend>
        <div
          aria-label="difficulty"
          class="MuiFormGroup-root MuiFormGroup-row"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-6"
                  name="difficulty"
                  type="radio"
                  value="easy"
                />
                <div
                  class="PrivateRadioButtonIcon-root-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Easy
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-6"
                  name="difficulty"
                  type="radio"
                  value="medium"
                />
                <div
                  class="PrivateRadioButtonIcon-root-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Medium
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-6"
                  name="difficulty"
                  type="radio"
                  value="hard"
                />
                <div
                  class="PrivateRadioButtonIcon-root-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Hard
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-4 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-6"
                  name="difficulty"
                  type="radio"
                  value=""
                />
                <div
                  class="PrivateRadioButtonIcon-root-7 PrivateRadioButtonIcon-checked-9"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Surprise Me!
            </span>
          </label>
        </div>
      </fieldset>
      <fieldset
        class="MuiFormControl-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          Type
        </legend>
        <div
          aria-label="type"
          class="MuiFormGroup-root MuiFormGroup-row"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-4 Mui-checked MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input-6"
                  name="type"
                  type="radio"
                  value="multiple"
                />
                <div
                  class="PrivateRadioButtonIcon-root-7 PrivateRadioButtonIcon-checked-9"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Multiple Choice
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-6"
                  name="type"
                  type="radio"
                  value="boolean"
                />
                <div
                  class="PrivateRadioButtonIcon-root-7"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-8"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              True/False
            </span>
          </label>
        </div>
      </fieldset>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Start New Game
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <p
      class="Feedback"
      data-testid="Feedback"
    />
  </div>
</DocumentFragment>
`;
